app/controllers/estimates_controller.rb

```diff
@@ -39,25 +39,11 @@ def download
       layout: "print_application",
     })
 
-    grover_options = {
-      format: "A4",
-      margin: {
-        top: "2cm",
-        bottom: "2cm",
-        left: "1cm",
-        right: "1cm",
-      },
-      emulate_media: "screen",
-      launch_args: ["--font-render-hinting=medium", "--no-sandbox"],
-      display_url: request.url.split("/estimates").first,
-      execute_script: "document.querySelectorAll('button').forEach(el => el.style.display = 'none')",
-    }
-
-    pdf = Grover.new(html, **grover_options).to_pdf
-
-    send_data pdf,
-              filename: "#{I18n.t('generic.download_name')} - #{Time.zone.now.strftime('%Y-%m-%d %H.%M.%S')}.pdf",
-              type: "application/pdf"
+    PdfService.with_pdf_data_from_html_string(html, request.url.split("/estimates").first) do |pdf_data|
```

Comment:
Is `request.original_url` a more up to date version of `request.url`?

Reply:
So `url` uses `fullpath` and `original_url` uses `original_fullpath`, and as far as I can tell, sometimes `fullpath` can be modified by Rails internals while `original_fullpath` is always left untouched. It doesn't really matter in this case because we're ignoring the path bit. So in fact we can just call `request.base_url` (which both `url` and `original_url` use), since that's the only bit we want anyway!